app/models/badges/all_your_base_badge.rb

```diff
@@ -5,7 +5,9 @@ class AllYourBaseBadge < Badge
       'all-your-base',
       'Completed the "All Your Base" exercise'
 
-    def award_to?(user)
+    def award_to?(user, exercise:)
+      return false unless exercise == 'all-your-base'
```

Comment:
The main downside of the current setup is that it requires the `exercise` keyword argument to be passed to the `AwardBadgeJob.perform_later` method call, which is not something that is obvious. We _could_ consider making the `exercise` keyword have a `nil` value by default, but I opted for not doing that 🤷 
Let me know what you think (pending this decision, I've not yet updated the unit tests, so they'll be failing).

Reply:
```
def worth_queuing?(context)
  context[:exercise] == 'all-your-base'
end
```

Reply:
```ruby
class AwardBadgeJob
  self.conditionally_perform_later(**args)
    return unless Badge.work_queuing?(**args)

    perform_later(**args*)
  end
end
```